### PR TITLE
Add some playwright ui tests

### LINKED ISSUE #20: Add playwright image tests
We need to add some playwright-based image tests to confirm it works properly in CI.

### PR DESCRIPTION
**Pull request by ianthomas23**
Add some playwright ui tests, following a similar approach to that used in `jupyterlite` and `pyodide-kernel` repositories. Currently there are tests for:

- loading and activating the extension
- opening a new terminal
- creating a new file
- visual image tests for an empty terminal and one after running a few commands

Note hiding file modification times so that tests are repeatable.

More tests will be added in future.

Fixes #20.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -106,14 +106,17 @@ jobs:
     - name: Install the extension
       run: |
         set -eux
+        python -m pip install --pre jupyterlite-core
```
Comment (jtpio) [inline .github/workflows/build.yml:109]: Maybe there could be a pin on `jupyterlite-core` here, as new versions may bring some visual changes that would make the UI tests fail?

For example `jupyterlite-core==0.5.0` will be based on JupyterLab 4.3 and Notebook 7.3, which may have some CSS changes compared to `jupyterlite-core==0.4.x`.

Comment (ianthomas23) [reply .github/workflows/build.yml:109]: Yes, I'll do that.

**Comment by ianthomas23** [comment]
please update snapshots

**Comment by ianthomas23** [comment]
please update snapshots

**Comment by ianthomas23** [comment]
I've moved the deployment-related files into their own `deploy` directory. Without this running `jupyter lite build` in the top-level directory finds `ui-tests/jupyter-lite.json` which causes problems. So now creating a deployment needs to be preceded by `cd deploy`.

I have tried to modify `vercel.json` to deal with this extra directory but I evidently haven't done this correctly and I cannot access the vercel build logs. @jtpio Could you take a look at this when you have some time? I think it should just involve changing a couple of the build settings in vercel rather than in the local `vercel.json` file.

**Comment by jtpio** [comment]
> I've moved the deployment-related files into their own `deploy` directory

Great, that's something I wanted to do at some point as a follow- up to https://github.com/jupyterlite/terminal/pull/28 :+1: 

> Without this running `jupyter lite build` in the top-level directory finds `ui-tests/jupyter-lite.json` which causes problems

Right, it looks related to this issue in JupyterLite: https://github.com/jupyterlite/jupyterlite/issues/1095

File: deploy/requirements-deploy.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+jupyterlab>=4,<5
+jupyterlite-core>=0.4,<0.5
+jupyterlite-pyodide-kernel
+.
```
Comment (jtpio) [inline deploy/requirements-deploy.txt:4]: Looks like we need to update to get the extension from the parent directory now:

```suggestion
..
```

**Comment by jtpio** [comment]
I updated the project settings to the following on Vercel:

image

**Comment by ianthomas23** [comment]
Thanks @jtpio, it is all working now.

**Pull request merged**